### ISSUE TITLE
[ISSUE] Issue with `databricks_permissions` resource: Tries to remove access with higher access instead of adding new permission

### ISSUE THREAD

**Issue by pcsrijith** [databricks/terraform-provider-databricks#3899]
### Configuration

```hcl
resource "databricks_group" "auto" {
  display_name = "Automation"
}

resource "databricks_directory" "this" {
  path = "/Workspace/Users/abc"
}

resource "databricks_permissions" "folder_usage" {
  directory_path = databricks_directory.this.path
  depends_on     = [databricks_directory.this]

  access_control {
    group_name       = databricks_group.auto.display_name
    permission_level = "CAN_RUN"
  }
}
```

### Expected Behavior

Groups to be added along with their respective permissions
### Actual Behavior

In Case, any group is already added with `CAN_MANAGE` access, it tries to remove that first before adding, resulting in failure.
### Steps to Reproduce

1. Find a folder which has `CAN_MANAGE` access (except "Admins") to some group.
2. Run the above code and add any new/existing group with lower access. For e.g. `CAN_RUN` access.
3. Terraform apply
```hcl
Error: cannot create permissions: Cannot remove Automation's CAN_MANAGE permission on 11111111111111
```

### Terraform and provider versions

Terraform Version: 1.9.3
Databricks provider version 1.49.1

### Would you like to implement a fix?

No

**Comment by alexott** [comment]
could be related to #3586

**Comment by alexott** [comment]
it's because `databricks_permissions` is authoritative resource - all permissions must be defined in one place. This blog post describes this: https://www.linkedin.com/posts/vuong-nguyen_here-is-something-that-came-up-quite-often-activity-7163117985031790593-9pHI?utm_source=share&utm_medium=member_desktop

There is a github issue to add `databricks_permission`, but someone should implement it.

**Comment by favoretti** [comment]
> could be related to #3586

Nope, broke on my dev provider with the patch merged as well. Will be looking into what happened tomorrow for us as well.

**Comment by Tanchwa** [comment]
> it's because `databricks_permissions` is authoritative resource - all permissions must be defined in one place. This blog post describes this: https://www.linkedin.com/posts/vuong-nguyen_here-is-something-that-came-up-quite-often-activity-7163117985031790593-9pHI?utm_source=share&utm_medium=member_desktop
> 
> There is a github issue to add `databricks_permission`, but someone should implement it.

Why is this the case? Terraform should be indepotent and always reference the code and state as the source of truth. Is this a limitation with the API or just the right hand not talking with the left?

**Comment by simonvanderveldt** [comment]
This seems to still be an issue.

In our case this is happening for SQL warehouse where we want to remove all permissions (Starter warehouse which we actually don't want at all, see https://github.com/databricks/terraform-provider-databricks/issues/3050#issuecomment-3167688357), which we can't because then it fails with "At least 1 "access_control" blocks are required." So we figured we'd just add the service principal we use for deploying but then it fails with
```
Error: cannot update permissions: cannot remove management permissions for the current user for warehouses, allowed levels: CAN_MANAGE, IS_OWNER
```
So apparently it's trying to remove the `IS_OWNER` permissions because we're trying to (also) give it `CAN_USE`.

The docs say "This resource is authoritative for permissions on objects" and not "This resource is authoritative for principals on objects", so that line in the docs is definitely not correct (or maybe it's incomplete?).

[edit] I now also see a Note stating
> If multiple permission levels are specified for an identity (e.g. CAN_RESTART and CAN_MANAGE for a cluster), only the highest level permission is returned and will cause permanent drift.

So this seems to be working as expected, although this is somewhat at odds with the "This resource is authoritative for permissions on objects" statement, so would at least be good to clean this discrepancy up in the docs.

**Comment by Tanchwa** [comment]
Hey I did a little bit of work on this a while ago and got the basic skeleton up. I want to keep working on it but I haven't billed less than 80 hours a week in the last two months. Give my name a search in the active PRs if you want to have a look; otherwise I'll take another crack at it when I have a breather.

**Comment by simonvanderveldt** [comment]
Thanks for the reply, but we do want an authoritative resource (well, we actually don't want the Starter warehouse, but since we always get it we want to ensure there are no permissions on it), but only as it's described in the docs (authoritative for permissions on objects) and not how it's implemented at the moment (authoritative for (some?) principals on objects).

**Comment by Tanchwa** [comment]
The way it works currently is that this resource `databricks_permissions` is authoritative in that a single run of this will overwrite any other rules in your configuration to a single resource weather they're in terraform or not.

The idea with `databricks_permission` is that it's not authoritative, and you can create as many of them as needed.

**Comment by lkanerv** [comment]
> I was wondering why it was designed this way at the first place.

+1 to this. It has been highly surprising to learn how many resources in this provider are authoritative and overwriting. Is that necessary?
